### DISCUSSION TITLE
Data Analysis/Mining Machine Learning Computer Vision S/W for OPI 5

### DISCUSSION THREAD

**Discussion by JayJay3** [Joshua-Riek/ubuntu-rockchip#1082] [General]
Fully GUI based powerful CMSR Data Mining / Machine Learning Studio is available for Orange Pi 5 users. It runs on Joshua Riek's Ubuntu and Armbian. (OPI's official ditros have webkit bugs. So you can't run on them currently.) It supports the following tools;

- Data analysis: powerful charts such as bars, pies, sankey, histogram, cross tables, grop-by reports, etc.
- Data mining: hotspot drill-down analysis, basket analysis, time-series, etc.
- Machine learning: ANN neural network, decision tree, Self Organizing Maps (SOM), regression, etc.
- Computer vision: Convolutional neural networks (CNN), object detection (YOLO tools), etc.
- Rule engine: SQL-like syntax rules, incorporating ANN, decision tree, SOM, regression models.

Minimum RAM size is 4GB. 8GB or more is recommended for large datasets.

Free download is available from the following link;

 [https://www.roselladb.com/downloadcmsform.php](https://www.roselladb.com/downloadcmsform.php)

The following link provides software information;

[ https://www.roselladb.com/starprobe.htm]( https://www.roselladb.com/starprobe.htm)

Except rule engine, all fully GUI based. No need for coding! It can turn your OPI 5 into data mining and machine learning system.

It is based on Java. So you can run on various machines. Supported systems are;
- Windows
- Linux (x64, RPI 4/5 Ubuntu, OPI 5 Joshua Riek's Ubuntu and Armbian)
- MacOS

It supports OpenCL GPU for computer vision tools. For large dataset computer vision, you will need powerful GPU. It works with SQL database systems such as PostgreSQL, MySQL, etc.

**Comment by MattiaGrigoli** [comment]
have you encountered any problem with the RGA 4G memory limit issue? if yes, how did you fixed it? (on the orange pi 5/rk3588)
